src/darjeeling/evaluator.py

```diff
@@ -138,12 +143,16 @@ def _filter_redundant_tests(self,
     def _run_test(self,
                   container: ProgramContainer,
                   candidate: Candidate,
-                  test: Test
+                  test: Test,
+                  heldout: bool = False
```

Comment:
minor: can we make `heldout` a keyword-only argument to improve readability at caller sites?
```python
test: Test,
*,
heldout: bool = False,
```